embedded/EscapeROOM/radio_morse_panel/source/misc.c

```diff
@@ -0,0 +1,88 @@
+#include <stdint.h>
+#include <stdbool.h>
+#include "globals.h"
+#include "misc.h"
+
+#define BTN_1_REL (RELEASED << PB0)
+#define BTN_2_REL (RELEASED << PB1)
+#define BTN_3_REL (RELEASED << PB2)
+#define BTN_4_REL (RELEASED << PB3)
+#define BTN_5_REL (RELEASED << PB4)
+#define BTN_6_REL (RELEASED << PB5)
+#define BTN_7_REL (RELEASED << PB6)
+#define BTN_8_REL (RELEASED << PB7)
+
+#define ALL_RELEASED	( BTN_1_REL | BTN_2_REL | BTN_3_REL | BTN_4_REL | \
+						BTN_5_REL | BTN_6_REL | BTN_7_REL | BTN_8_REL )
+
+enum ButtonState { PRESSED, RELEASED };
+
+typedef struct {
+	unsigned char pin : 3;
+	char keyChar;
+} Button;
+
+static Button buttons[] = {
+	{ PB0, '1'},
+	{ PB1, '2'},
+	{ PB2, '3'},
+	{ PB3, '4'},
+	{ PB4, '5'},
+	{ PB5, '6'},
+	{ PB6, '7'},
+	{ PB7, '8'}
+};
+
+bool isOneBitSet (uint8_t x) {
+	return ( x && !(x & (x - 1)) ) ? true : false;
+}
+
+int8_t getSetBitPosition (uint8_t x) {
+	uint8_t pos = 0;
+
+	if (x == 0) {
+		return -1;
+	}
+	
+	while ( x != 1 ) {
+		x >>= 1;
+		pos++;
+	}
+	return pos;
+}
+
+uint32_t getDeltaTime (uint32_t t1, uint32_t t0) {
+	return t1 - t0;
+}
+
+void updateButtonState(void) {
```

Comment:
Following comment is my personal preference, not a bug or change request.
In misc.c I thend to put small helper functions, that does not belong to any specific object and are re-used from several other .c files. Like isOneBitSet() or getSetBitPosition().
updateButtonState() is not a helper in my opinion.  It belongs to object (.c file ) that deals with buttons and/or user input. I would put all button related functions into separate file - button.c.

Reply:
Taking this into account in future. Thanks.